### ISSUE TITLE
Option currying for consistency with the official next plugins

### ISSUE THREAD

**Issue by kachkaev** [martpie/next-transpile-modules#52]
**Is your feature request related to a problem? Please describe.**

At the moment, the plugin expects Next.js config to contain `transpileModules` option. There seems to exist a tendency towards passing options via currying and thus avoiding the pollution of the Next.js config.

You can see this by comparing [`@zeit/next-bundle-analyzer`](https://github.com/zeit/next-plugins/blob/9ecca7b492518464897ea866b4abcdc6214bf705/packages/next-bundle-analyzer/index.js) (deprecated) and [`@next/bundle-analyzer`](https://github.com/zeit/next.js/blob/canary/packages/next-bundle-analyzer/index.js) (current). To me as to a user of [`next-compose-plugins`](https://www.npmjs.com/package/next-compose-plugins) this means:

```diff
 // next.config.js
 module.exports = require("next-compose-plugins")([
-  [require("@zeit/next-bundle-analyzer"), { analyzeServer: true, analyzeClient: true }),
+  require("@next/bundle-analyzer")({ enabled: true }),
   require("@zeit/next-source-maps")(),
   [require("next-transpile-modules"), { transpileModules: ["a", "b", "c"] }],
])
```

As you can see, `next-transpile-modules` somewhat stands out now because it does not use currying. Note `[ ... ]` on that line.

**Describe the solution you'd like**

I'd like to see currying in `next-transpile-modules` as well, which would make plugins look more similar and avoid pollution of the Next.js config object. It will make plugin composition when producing combined plugins easier as well.

```diff
 // next.config.js
 module.exports = require("next-compose-plugins")([
   require("@next/bundle-analyzer")({ enabled: true }),
   require("@zeit/next-source-maps")(),
-  [require("next-transpile-modules"), { transpileModules: ["a", "b", "c"] }],
+  require("next-transpile-modules")({ modules: ["a", "b", "c"] }),
])
```

Another example to demonstrate the idea:

```js
const withBundleAnalyzer = require("@zeit/next-bundle-analyzer");
const withTM = require('next-transpile-modules');

module.exports = withBundleAnalyzer(
  withTM({
    analyzeServer: true,
    analyzeClient: true,
    transpileModules: ['somemodule', 'and-another'],
    someNextJsOption: 42,
  })
);
```
↓
```js
const withBundleAnalyzer = require("@next/bundle-analyzer")({ enabled: true });
const withTM = require('next-transpile-modules')({ modules: ["a", "b", "c"] });

module.exports = withBundleAnalyzer(
  withTM({
    someNextJsOption: 42,
  })
);
```

**PS:** It might be worth updating `README` as well given the opportunity. It refers to `@zeit/next-typescript`, which has been deprecated a few months back following the release of Next.js 9.

**Comment by martpie** [comment]
Thank you very much for opening this issue, I'll investigate how Zeit manages to do it for its current plugins :)

I should definitely update the FAQ as well :)

**Comment by martpie** [comment]
This is actually a breaking change (just noticed). It is really odd I did not notice this syntax before.

v3.0 on its way.

**Comment by kachkaev** [comment]
Yes, that's a breaking change indeed. Wondering if it's possible to gracefully fail with a reasonable explanation if devs still use the old function style. That'd be cool!

> It is really odd I did not notice this syntax before.

I think the preferred syntax changed just a few months ago, I remember getting a bit surprised seeing the new API of `@next/bundle-analyzer`. Makes more sense that the original flavour though!

**Comment by martpie** [comment]
Fixed in cb86a25 

will be released when the rest of the 3.0 milestone is done
